src/fastp/config.vsh.yaml

```diff
@@ -0,0 +1,541 @@
+        - for PE data, you should also specify read2 input by `-I` or `--in2`.
+      arguments:
+        - name: --in1
+          alternatives: [-i]
+          type: file
+          description: Input FastQ file. Must be single-end or paired-end R1. Can be gzipped.
+          required: true
+          example: in.R1.fq.gz
+        - name: --in2
+          alternatives: [-I]
+          type: file
+          description: Input FastQ file. Must be paired-end R2. Can be gzipped.
+          required: false
+          example: in.R2.fq.gz
+    - name: Outputs
+      description: |
+
+        - for SE data, you only have to specify read1 output by `-o` or `--out1`.
+        - for PE data, you should also specify read2 output by `-O` or `--out2`.
+        - if you don't specify the output file names, no output files will be written, but the QC will still be done for both data before and after filtering.
+        - the output will be gzip-compressed if its file name ends with `.gz`
+      arguments:
+        - name: --out1
+          alternatives: [-o]
+          type: file
+          description: The single-end or paired-end R1 reads that pass QC. Will be gzipped if its file name ends with `.gz`.
+          required: true
+          example: out.R1.fq.gz
+          direction: output
+        - name: --out2
+          alternatives: [-O]
+          type: file
+          description: The paired-end R2 reads that pass QC. Will be gzipped if its file name ends with `.gz`.
+          required: false
+          example: out.R2.fq.gz
+          direction: output
+        - name: --unpaired1
+          type: file
+          description: Store the reads that `read1` passes filters but its paired `read2` doesn't.
+          required: false
+          example: unpaired.R1.fq.gz
+          direction: output
+        - name: --unpaired2
+          type: file
+          description: Store the reads that `read2` passes filters but its paired `read1` doesn't.
+          required: false
+          example: unpaired.R2.fq.gz
+          direction: output
+        - name: --failed_out
+          type: file
+          description: |
+            Store the reads that fail filters.
+
+            If one read failed and is written to --failed_out, its failure reason will be appended to its read name. For example, failed_quality_filter, failed_too_short etc.
+            For PE data, if unpaired reads are not stored (by giving --unpaired1 or --unpaired2), the failed pair of reads will be put together. If one read passes the filters but its pair doesn't, the failure reason will be paired_read_is_failing.
+          required: false
+          example: failed.fq.gz
+          direction: output
+        - name: --overlapped_out
+          type: file
+          description: |
+            For each read pair, output the overlapped region if it has no any mismatched base.
+          direction: output
+    - name: Report output arguments
+      arguments:
+        - name: --json
```

Comment:
Add defaults?

Reply:
A user may or may not want a report as an output, so I'd say not adding defaults for --json and --html is the way to go

Reply:
Ok, just wanted to make sure that we are deviating from what fastp does by default.
```
 -j, --json                           the json format report file name (string [=fastp.json])
 -h, --html                           the html format report file name (string [=fastp.html])
```